streams/src/test/java/org/apache/kafka/streams/integration/EosBetaUpgradeIntegrationTest.java

```diff
@@ -261,24 +241,24 @@ public void shouldUpgradeFromEosAlphaToEosBeta() throws Exception {
             streams1Alpha.setStateListener(
                 (newState, oldState) -> stateTransitions1.add(KeyValue.pair(oldState, newState))
             );
+            int expectedNumStableAssignments = assignmentListener.numStableAssignments() + 1;
             streams1Alpha.cleanUp();
             streams1Alpha.start();
-            waitForStateTransition(
-                stateTransitions1,
-                Arrays.asList(
-                    KeyValue.pair(KafkaStreams.State.CREATED, KafkaStreams.State.REBALANCING),
-                    KeyValue.pair(KafkaStreams.State.REBALANCING, KafkaStreams.State.RUNNING)
-                )
-            );
+            assignmentListener.waitForNextStableAssignment(expectedNumStableAssignments, MAX_WAIT_TIME_MS);
```

Comment:
So far we _always_ pass in `assignmentListener.numStableAssignments() + 1` -- hence, I am wondering if we need to pass it at all?

Reply:
Well we need to record it _before_ doing whatever action causes the group to rebalance. But maybe we can do something like `prepareForRebalance` if that seems easier?

Reply:
I see. Guess it's good as-is.